### ISSUE TITLE
windows os

### ISSUE THREAD

**Issue by Reaper10** [nixxcode/amsynth#2]
am synth needs to come windows

**Comment by mcanthony** [comment]
I am porting this to PNaCl (Portable Native Client) which will enable it to be used in Chromium/CEF runtimes such as Chrome, Opera, NW.JS, or Atom/Electron, as well as an Emscripten generated build in ASM.JS for other browsers. Based on what I have seen so far in the code however, this is likely ready to build in Windows with little to no modification, so @Reaper10 since Visual Studio has a free version and MSBuild is now even Open Source you should be equipped to build yourself a shiny new amsynth32.exe/

**Comment by koerg** [comment]
did anyone have success in building a VST plugin for windows? I am currently downloading/configuring Visual Studio. It takes ages ...
Is it possible at all to build a dll-file with current dependencies?

**Comment by nixxcode** [comment]
Yes; there is an experimental `win32` branch containing a Visual Studio 2015 project and some code changes necessary to build on Windows. Note that it lacks a GUI or loading of preset banks.

**Comment by Moth-Tolias** [comment]
the "win32" branch seems to be missing - is this still being worked on?

**Comment by nickdowell** [comment]
The `win32` was merged into `develop` - to build the Windows VST version, open `vs/amsynth.sln`

Note that it's still pretty basic, lacking a GUI or loading of preset banks.

**Comment by AnClark** [comment]
![image](https://user-images.githubusercontent.com/12870704/126856005-a1e5c0f8-5902-4b0a-aafa-b921220dd72f.png)

Here's my implementation. Still very experimental and unstable.

**Comment by nickdowell** [comment]
> Here's [my implementation](https://github.com/anclark/amsynth). Still very experimental and unstable.

Woah! The GTK UI works in REAPER on Windows?!

**Comment by AnClark** [comment]
> Woah! The GTK UI works in REAPER on Windows?!

Yes, it works!
But there are still a lot of things to do, for example, to let it embed properly, and to fix several crashes.
I still prefer rewritting the UI with other frameworks.

**Comment by AnClark** [comment]
@kmturley Currently GTK on Windows is unstable and deprecated (you can see my screenshot above). So I decided to rewrite it with Dear ImGui. Still work in progress, but has a ready-to-use simple UI.

You can refer to my fork: https://github.com/anclark/amsynth (I'll write a build guide these days).

**Comment by nickdowell** [comment]
After rewriting the GUI in JUCE (see #37) it was quite easy to get amsynth building for Windows (VST2) with full GUI including preset menus, High DPI support... 😁 

<img width="866" alt="Capture" src="https://github.com/user-attachments/assets/74e757d4-9528-4d3e-9d96-6012bcc8b898">

I still have not made an official release with the rewritten UI, so for anyone interested in a preview here's an installer for 64-bit Windows - [amsynth_1.14.0.exe (Dropbox)](https://www.dropbox.com/scl/fi/98mhiu8n23ikizi3zhllc/amsynth_1.14.0.exe?rlkey=cy2ab0x30cc7f0b9lj78e8pdf&st=n9qirskg&dl=0). I don't plan to support 32-bit Windows unless there is demand.

If you don't want to run an installer from a random-looking Dropbox link, you can build the plug-in using the Visual Studio solution in the [`win` folder](https://github.com/amsynth/amsynth/tree/develop/win) but note that the plug-in requires that the banks and skins directories are installed to `C:\ProgramData\amsynth`

**Comment by AnClark** [comment]
@nickdowell Great! Using JUCE is a brilliant idea. Now amsynth is really cross-platform thanks to JUCE.

**Comment by nickdowell** [comment]
Resolved in https://github.com/amsynth/amsynth/releases/tag/release-2.0.0
